### PR TITLE
ocaml-pcre2: new port 7.5.2

### PR DESCRIPTION
**Pull request by pguyot**
###### Tested on

macOS 14.4.1 23E224 x86_64
Xcode 15.3 15E204a

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?

### CODE REVIEW

File: ocaml/ocaml-pcre2/Portfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+description         Bindings to the Perl Compatibility Regular Expressions library (version 2)
+long_description    pcre2-ocaml offers library functions for string pattern \
+                    matching and substitution, similar to the functionality \
+                    offered by the Perl language.
+
+homepage            https://github.com/camlp5/pcre2-ocaml
+
+checksums           md5     f52d4df71fc67e7faa3061d2feac96e4 \
+                    sha1    a4b8536c01f7b8001981b7eab6a0a22d67362161 \
```
Comment (barracuda156) [inline ocaml/ocaml-pcre2/Portfile:27]: Same, please remove md5 and sha1.

File: ocaml/ocaml-pcre2/Portfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+version             7.5.2
+github.setup        camlp5 pcre2-ocaml ${version}
+github.tarball_from releases
+distname            pcre2-${version}
+extract.suffix      .tbz
+revision            0
+categories          ocaml devel
+maintainers         {pguyot @pguyot} openmaintainer
+license             LGPL
+platforms           darwin
```
Comment (barracuda156) [inline ocaml/ocaml-pcre2/Portfile:17]: `platforms darwin` is the default value, we only need to specify platforms if those are different from all Darwin.

**Comment by barracuda156** [comment]
Extract is broken:
```
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_ocaml_ocaml-pcre2/ocaml-pcre2/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/ocaml-pcre2/pcre2-7.5.2.tbz' | /usr/bin/gnutar --no-same-owner -xf - 

gzip: /opt/local/var/macports/distfiles/ocaml-pcre2/pcre2-7.5.2.tbz: not in gzip format
--->  Configuring ocaml-pcre2
--->  Building ocaml-pcre2
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_ocaml_ocaml-pcre2/ocaml-pcre2/work/pcre2-7.5.2" && /opt/local/bin/dune build -j 6 --root=/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_ocaml_ocaml-pcre2/ocaml-pcre2/work/pcre2-7.5.2 --profile=release --ignore-promoted-rules --no-config --only-packages=pcre2 --default-target=@install @install 
Error: I don't know about package pcre2 (passed through --only-packages)
Command failed:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_ocaml_ocaml-pcre2/ocaml-pcre2/work/pcre2-7.5.2" && /opt/local/bin/dune build -j 6 --root=/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_ocaml_ocaml-pcre2/ocaml-pcre2/work/pcre2-7.5.2 --profile=release --ignore-promoted-rules --no-config --only-packages=pcre2 --default-target=@install @install 
Exit code: 1
Error: Failed to build ocaml-pcre2: command execution failed
```
We need `use_bzip2 yes` in the portfile.

**Comment by ryandesign** [comment]
> Extract is broken:

Only on OS X 10.8 and earlier.

> We need `use_bzip2 yes` in the portfile.

Correct.

File: ocaml/ocaml-pcre2/Portfile (hunk 2 above)
Comment (pguyot) [reply ocaml/ocaml-pcre2/Portfile:17]: Man page says it is required, should we update it?

Comment (ryandesign) [reply ocaml/ocaml-pcre2/Portfile:17]: Yes.

Comment (pguyot) [reply ocaml/ocaml-pcre2/Portfile]: Thanks.
I created a PR for this: [#macports-base/pull/326](https://github.com/macports/macports-base/pull/326)

**Comment by barracuda156** [comment]
@pguyot Otherwise both ports build fine on powerpc for me. Both errors were just port extract not being right.

**Comment by pguyot** [comment]
> @pguyot Otherwise both ports build fine on powerpc for me. Both errors were just port extract not being right.

Thanks for testing. These two ports would let us upgrade and fix camlp5 port which is currently broken on Sonoma.

File: ocaml/ocaml-pcre2/Portfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+PortSystem          1.0
+PortGroup           ocaml 1.1
+PortGroup           github 1.0
+
+name                ocaml-pcre2
+version             7.5.2
+revision            0
+categories          ocaml devel
+maintainers         {pguyot @pguyot} openmaintainer
+license             LGPL
```
Comment (reneeotten) [inline ocaml/ocaml-pcre2/Portfile:12]: this should be `LGPL-2.1`

File: ocaml/ocaml-pcre2/Portfile [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                    offered by the Perl language.
+
+homepage            https://github.com/camlp5/pcre2-ocaml
+
+checksums           rmd160  8931ef2028de737b262b676bb48934301c6951a5 \
+                    sha256  da3b9a6e566707c64ab78468fb4c453e930716740e3ae911566f6a378d0562c8 \
+                    size    35073
+
+depends_lib-append  port:pcre2
+depends_build-append  port:ocaml-dune-configurator
```
Comment (reneeotten) [inline ocaml/ocaml-pcre2/Portfile:32]: this isn't aligned with four spaces

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: ocaml/ocaml-pcre2/Portfile (hunk 3 above)
Comment (pguyot) [reply ocaml/ocaml-pcre2/Portfile]: Thanks. I couldn't find a list of licenses, maybe we should stick to SPDX?

Comment (reneeotten) [reply ocaml/ocaml-pcre2/Portfile]: no, we should use a license that MacPorts recognized as it is used to decide whether a port is distributable (see: https://github.com/macports/macports-infrastructure/blob/d467a2b8e417742baf678042f9f93c392a45dcac/jobs/distributable_lib.tcl).

Comment (pguyot) [reply ocaml/ocaml-pcre2/Portfile]: Thanks. I searched for the list in base. I'll add it to port lint eventually.

**Pull request merged**